nanofi/include/ecu_api/ecuapi.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+    *
+    *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#ifndef NIFI_MINIFI_CPP_ECUAPI_H
+#define NIFI_MINIFI_CPP_ECUAPI_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "uthash.h"
+#include <core/threadpool.h>
+#include <core/message_queue.h>
+#include <coap/c2structs.h>
+
+typedef enum io_type {
+    SITE2SITE,
+    TAILFILE,
+    KAFKA,
+    MQTTIO,
+    MANUAL
+} io_type_t;
+
+typedef struct ecu_context {
+    char * name;
+    char uuid[37];
+    io_type_t input;
+    io_type_t output;
+    //configuration parameters from configuration file
+    //contains, c2 host, c2 port, c2 heartbeat uri
+    //c2 acknowledge uri, heartbeat interval
+    properties_t * ecu_configuration;
+    void * input_processor_ctx;
+    void * output_processor_ctx;
+    threadpool_t * thread_pool;
```

Comment:
is thread pool something that is configurable? ideally we create no threads and be entirely synchronous if desired. I think the default should be async but we should also support synchronous interactions.

Reply:
Yes, we have an api that supports synchronous interactions if the user needs to ingest data from within the application using the sdk and that is not yet integrated with c2. It is still a wip. I will make the threadpool configurable.

Reply:
We have a synchronous api for ingesting data from within the application. However, that is not yet integrated into c2. Threadpool is not configurable yet. I will include it with the c2 configuration parameters when I am working on it.